digiwf-integrations/digiwf-email-integration/digiwf-email-integration-core/src/main/java/de/muenchen/oss/digiwf/email/integration/adapter/in/MessageProcessor.java

```diff
@@ -18,29 +20,63 @@
 public class MessageProcessor {
 
     private final ErrorApi errorApi;
-    private final SendMail mailUseCase;
+    private final SendMailInPort mailUseCase;
     private final MonitoringService monitoringService;
 
-    public Consumer<Message<Mail>> emailIntegration() {
+    public Consumer<Message<TextMail>> emailIntegration() {
+
         return message -> {
-            try {
-                this.mailUseCase.sendMail(
+            withErrorHandling(message, () -> {
+                this.mailUseCase.sendMailWithText(
                         message.getHeaders().get(DIGIWF_PROCESS_INSTANCE_ID, String.class),
                         message.getHeaders().get(TYPE, String.class),
                         message.getHeaders().get(DIGIWF_INTEGRATION_NAME, String.class),
                         message.getPayload());
                 this.monitoringService.sendMailSucceeded();
-            } catch (final BpmnError bpmnError) {
-                this.monitoringService.sendMailFailed();
-                this.errorApi.handleBpmnError(message.getHeaders(), bpmnError);
-            } catch (final ValidationException validationException) {
-                this.monitoringService.sendMailFailed();
-                this.errorApi.handleBpmnError(message.getHeaders(), new BpmnError("VALIDATION_ERROR", validationException.getMessage()));
-            } catch (final IncidentError incidentError) {
-                this.monitoringService.sendMailFailed();
-                this.errorApi.handleIncident(message.getHeaders(), incidentError);
-            }
+            });
+        };
+    }
+
+    public Consumer<Message<MailWithLogoAndLinkDto>> sendMailWithLogoAndLink() {
+
+        return message -> {
+            MailWithLogoAndLinkDto mail = message.getPayload();
+            withErrorHandling(message, () -> {
+                this.mailUseCase.sendMailWithTemplate(
```

Comment:
```suggestion
            withErrorHandling(message, () -> {
                MailWithLogoAndLinkDto mail = message.getPayload();
                this.mailUseCase.sendMailWithTemplate(
```

Sonst wird die Validierung nicht gefangen